pkg/deploy/generator/resources_rp.go

```diff
@@ -456,8 +468,19 @@ func (g *generator) rpVMSS() *arm.Resource {
 			},
 			Tags: map[string]*string{},
 			VirtualMachineScaleSetProperties: &mgmtcompute.VirtualMachineScaleSetProperties{
+				// Reference: https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-automatic-upgrade#arm-templates
 				UpgradePolicy: &mgmtcompute.UpgradePolicy{
-					Mode: mgmtcompute.UpgradeModeRolling,
+					Mode: mgmtcompute.UpgradeModeAutomatic,
+					RollingUpgradePolicy: &mgmtcompute.RollingUpgradePolicy{
+						// Percentage equates to 1.02 instances out of 3
+						MaxBatchInstancePercent: to.Int32Ptr(34),
+						MaxUnhealthyInstancePercent: to.Int32Ptr(34),
+						MaxUnhealthyUpgradedInstancePercent: to.Int32Ptr(34),
+						PauseTimeBetweenBatches: to.StringPtr("PT10M"),
+					},
+					AutomaticOSUpgradePolicy: &mgmtcompute.AutomaticOSUpgradePolicy{
+						EnableAutomaticOSUpgrade: to.BoolPtr(true),
+					},
```

Comment:
It appears that our sdk version doesn't have that field. [azure-sdk-for-go@v63.1.0 - AutomaticOSUpgradePolicy](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v63.1.0+incompatible/services/compute/mgmt/2020-06-01/compute#AutomaticOSUpgradePolicy)

Reply:
Oops, I didn't realize our SDK version doesn't have that field. It's not super clear whether the rolling upgrade policy will be used for OS updates then... it's probably fine for now since moving to the track 2 SDK is out of scope here.

I think it's already covered in Jira: https://issues.redhat.com/browse/ARO-4661